python/rand_forest.py

```diff
@@ -130,72 +169,187 @@ def run():
     # num_iter-1 generations since the initial population is evaluated once first
     start_time = time.time()
 
+    evaluated_points_from_file = {}
+
+    file_to_dict(previous_results_file,evaluated_points_from_file)
 
-    # Initialization of dataset  
+    #dictionary to save already evaluated points of this run and their results
+    evaluated_points = {}
 
-    x = make_random_points(num_init_population) 
-    evals = evaluate(x)
-    y = (list(map(lambda v : 0 if v>0 else 1, evals)))
+    # Initialization of dataset
+    x = make_random_points(num_init_population)
 
+    x_new = [point for point in x if (str(point) not in evaluated_points_from_file)]
```

Comment:
If the above is turned to an optional step, then some changes will be required here too.